client/my-sites/themes/current-theme/index.jsx

```diff
@@ -88,7 +93,13 @@ const CurrentThemeWithOptions = ( { siteId, currentTheme } ) => (
 );
 
 export default connect(
-	( state, { siteId } ) => ( {
-		currentTheme: getCurrentTheme( state, siteId )
-	} )
+	( state, { siteId } ) => {
+		const currentThemeId = getActiveTheme( state, siteId );
+		const siteIdOrWpcom = isJetpackSite( state, siteId ) ? siteId : 'wpcom';
+		return {
+			currentThemeId,
+			currentTheme: getTheme( state, siteIdOrWpcom, currentThemeId ),
```

Comment:
Probably a bit late with this, but would it make sense to have `siteId` as the third parameter to `getTheme()`, defaulting to `'wpcom'`?